mitodl/mit-learn
# Duration and time commitment

### What are the relevant tickets?
Partially implements https://github.com/mitodl/hq/issues/5130 and https://github.com/mitodl/hq/issues/5131

### Description (What does it do?)
- Adds the following fields to `LearningResourceRun` and `LearningResource`:
  - `duration`  (string)
  - `min_weeks`  (int)
  - `max_weeks` (int)
  - `time_commitment` (string)
  -  `min_weekly_hours` (int)
  -  `max_weekly_hours`
- Populates these values for `LearningResourceRun` during ETL pipeline runs for:
  - MITx Online 
  - xPro (relevant fields in the xPro API should be present within a few days)
  - MIT edX
- Populates the corresponding values for `LearningResource` based on the values of the next/best run.


### How can this be tested?
Add the following to your .env file using RC values
   ```
   EDX_*
   MITX_ONLINE_*
   XPRO_*
   ```
Run the following, they should all complete successfully, and when finished,  at least some programs and courses from each source (except xPro) should have values for the new fields:
  - `./manage.py backpopulate_mitxonline_data`
  - `./manage.py backpopulate_oll_data`
  - `./manage.py backpopulate_mit_edx_data`
  - `./manage.py backpopulate_xpro_data`

### Additional Context
Some sources provide this information at the run level (MIT edX, Sloan, Professional Education), others at the course/program level.  So I decided to populate the data for runs across all sources in the ETL pipelines, and then populate values at the course/program level based on the next/best run (via the `learning_resources.etl.loaders.load_run_dependent_values` function).

The Sloan and Professional Education ETL pipelines will be updated in a subsequent PR.  They will require lots more regex shenanigans to calculate min/max duration and commitment values from free-form strings.
